src/main/java/seedu/exercise/logic/commands/CommandResult.java

```diff
@@ -48,6 +56,46 @@ public String getFeedbackToUser() {
         return feedbackToUser;
     }
 
+    /**
+     * Checks that {@code CommandResult} is only showing at most 1 resource list
+     * and sets {@code showExerciseList} to be true
+     */
+    public CommandResult setShowExerciseList() {
+        assert !(showExerciseList || showRegimeList || showScheduleList || showSuggestionlist);
+        showExerciseList = true;
+        return this;
+    }
+
+    /**
+     * Checks that {@code CommandResult} is only showing at most 1 resource list
+     * and sets {@code showRegimeList} to be true
+     */
+    public CommandResult setShowRegimeList() {
+        assert !(showExerciseList || showRegimeList || showScheduleList || showSuggestionlist);
+        this.showRegimeList = true;
+        return this;
+    }
+
+    /**
+     * Checks that {@code CommandResult} is only showing at most 1 resource list
+     * and sets {@code showScheduleList} to be true
+     */
+    public CommandResult setShowScheduleList() {
+        assert !(showExerciseList || showRegimeList || showScheduleList || showSuggestionlist);
+        this.showScheduleList = true;
+        return this;
+    }
+
+    /**
+     * Checks that {@code CommandResult} is only showing at most 1 resource list
+     * and sets {@code showSuggestionList} to be true
+     */
+    public CommandResult setShowSuggestionList() {
+        assert !(showExerciseList || showRegimeList || showScheduleList || showSuggestionlist);
+        this.showSuggestionlist = true;
+        return this;
+    }
+
```

Comment:
If the code shows some builder pattern, should extend it to the rest of the variables too. In the builder, you can then ensure only one of these methods are called.

Reply:
ok